### PR TITLE
Add email, date and phone number fields 

### PR DESCRIPTION
**Pull request by EmiPali**
Three components are added, email field, phone number field and date.

Screenshot 2024-06-26 at 15 50 10
Screenshot 2024-06-26 at 16 08 15

### CODE REVIEW

**Comment by sulejmank** [comment]
I don't see you using these fields for Application form?

**Comment by EmiPali** [comment]
@sulejmank sorry my bad, I thought we needed just the components for the moment. Will integrate them to the Application form now.

**Comment by sulejmank** [comment]
No worries, also to test this properly, you will need to change form on REMS for some specific dataset. So basically on rems side is defined how form looks, and we then just based on rems response display components needed. Check my previous PR

**Comment by EmiPali** [comment]
@sulejmank you can check It now.

File: package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "next-auth": "^4.24.5",
     "next-runtime-env": "^3.2.0",
     "react": "^18",
+    "react-datepicker": "^7.2.0",
```
Comment (brunopacheco1) [inline package.json:39]: suggestion: If it is not too much work, perhaps you could replace react-datepicker to [shadcn date picker](https://ui.shadcn.com/docs/components/date-picker), to stick to a single component lib provider.

**Review by brunopacheco1 — APPROVED** [review]

File: src/app/applications/[id]/DateField.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  return (
+    <div className="rounded border p-4">
+      <div className="flex flex-col justify-between">
+        <div>
+          <h3 className="text-lg text-primary sm:text-xl">{`${title} ${field.optional ? "(Optional)" : ""}`}</h3>
+        </div>
+        <DatePicker
+          selected={inputValue}
+          onChange={handleDateChange}
+          className="mt-4 h-12 w-full rounded-lg border-2 border-primary px-4 py-[9px] text-base shadow-sm transition-all duration-200 ease-in-out hover:shadow-md focus:border-transparent focus:outline-none focus:ring-2 focus:ring-primary"
```
Comment (inderps) [inline src/app/applications/[id]/DateField.tsx:49]: Suggestion: You can think of moving these input related classes to a constant and then re-use it everywhere

**Review by inderps — APPROVED** [review]
Looks good with one comment about using shadcdn for date

**Review by sulejmank — APPROVED** [review]

**Pull request merged**